tutorials/throttle/src/main/java/com/datatorrent/examples/throttle/ThrottlingStatsListener.java

```diff
@@ -0,0 +1,139 @@
+            logger.info("Sending suspend request");
+            List<OperatorRequest> operatorRequests = new ArrayList<OperatorRequest>();
+            operatorRequests.add(new InputSlowdownRequest());
+            response.operatorRequests = operatorRequests;
+            */
+            logger.info("Setting suspend");
+            normalState = false;
+        } else if (!normalState && ((maxWindow - minWindow) <= windowThreshold)) {
+            // In future send the request here which will happen only on a state change
+            /*
+            // Send request to operator to get back to normal
+            logger.info("Sending normal request");
+            List<OperatorRequest> operatorRequests = new ArrayList<OperatorRequest>();
+            operatorRequests.add(new InputNormalRequest());
+            response.operatorRequests = operatorRequests;
+            */
+            logger.info("Setting normal");
+            normalState = true;
+        }
+
+        if (!normalState) {
+            // Send request to operator to slow down
+            List<OperatorRequest> operatorRequests = new ArrayList<OperatorRequest>();
+            operatorRequests.add(new InputSlowdownRequest());
+            response.operatorRequests = operatorRequests;
+        } else {
+            // Send request to operator to get back to normal
+            List<OperatorRequest> operatorRequests = new ArrayList<OperatorRequest>();
+            operatorRequests.add(new InputNormalRequest());
+            response.operatorRequests = operatorRequests;
+        }
+
+        return response;
+    }
+
+    // This runs on the operator side
+    public static class InputSlowdownRequest implements OperatorRequest, Serializable
+    {
+        private static final Logger logger = LoggerFactory.getLogger(InputSlowdownRequest.class);
+
+        @Override
+        public OperatorResponse execute(Operator operator, int operatorId, long windowId) throws IOException
+        {
+            logger.debug("Receieved slowdown operator {} operatorId {} windowId {}", operator, operatorId, windowId);
+            if (operator instanceof RandomNumberGenerator) {
+                RandomNumberGenerator generator = (RandomNumberGenerator)operator;
+                generator.suspend();
+            }
+            return new InputOperatorResponse();
+        }
+    }
+
+    public static class InputNormalRequest implements OperatorRequest, Serializable
+    {
+        @Override
+        public OperatorResponse execute(Operator operator, int operatorId, long windowId) throws IOException
+        {
+            logger.debug("Receieved normal operator {} operatorId {} windowId {}", operator, operatorId, windowId);
+            if (operator instanceof RandomNumberGenerator) {
+                RandomNumberGenerator generator = (RandomNumberGenerator)operator;
+                generator.normal();
+            }
+            return new InputOperatorResponse();
+        }
+    }
```

Comment:
It is a good idea but there could be other differing arguments that could be sent in future for slowdown vs the normal (like degree of slowdown etc). It is probably ok to leave it this way as illustration that it is possible to send different types of requests with varying information.